xdev_bot/issues.py

```diff
@@ -8,79 +8,74 @@
+        card_id = df.loc[row]['card_id'].values[0]
 
+        if event.data['action'] == 'closed':
+            card_action = 'Closing'
+            column_id = PROJECT_BOARD['columns']['done']['id']
 
-    """
-    issue_url = event.data['issue']['html_url']
-    project_board_name = PROJECT_BOARD['name']
-    df = read_database()
-    row = df['note'] == issue_url
-    card_id = df.loc[row]['card_id'].values[0]
-    url = f'/projects/columns/cards/{card_id}/moves'
+            if event.data[0] == 'pull_request':
+                dict_of_labels = event.data['pull_request']['labels']
+                labels = []
 
-    if event.data['action'] == 'closed':
-        done_column_id = PROJECT_BOARD['columns']['done']['id']
-        print(f'Updating database: move card {card_id} to done column')
+                for item in dict_of_labels: labels.append(item['name'])
+                labels = set(labels)
 
-        print(f'Closing Card in {project_board_name} project board for issue : {issue_url}')
-        # POST /projects/columns/cards/:card_id/moves
+                if 'needs-review' in labels: labels.remove('needs-review')
 
-        await gh.post(
-            url,
-            data={'position': 'top', 'column_id': done_column_id},
-            accept='application/vnd.github.inertia-preview+json',
-        )
+                merged = event.data['pull_request']['merged']
+                if merged: labels.add('merged')
+                else: labels.add('rejected')
+        
+                labels = list(labels)
+                await gh.patch(issue_url, data={'labels': labels})
 
-    else:
-        in_progress_column_id = PROJECT_BOARD['columns']['in_progress']['id']
-        print(f'Updating database: move card {card_id} to in-progress column')
+        elif event.data['action'] == 'reopened':
+            card_action = 'Reopening'
+            column_id = PROJECT_BOARD['columns']['in_progress']['id']
 
-        await gh.post(
-            url,
-            data={'position': 'top', 'column_id': in_progress_column_id},
-            accept='application/vnd.github.inertia-preview+json',
-        )
-        print(f' Reopening card in {project_board_name} project board for issue : {issue_url}')
+        url = f'/projects/columns/cards/{card_id}/moves'
 
+    await gh.post(
+        url,
+        data={'position': 'top', 'column_id': column_id},
+        accept='application/vnd.github.inertia-preview+json')
+    print(f' {card_action} card in {project_board_name} project board for issue : {html_url}')
 
 @router.register('issues', action='assigned')
 @router.register('issues', action='unassigned')
-async def issue_assigned_event(event, gh, *args, **kwargs):
-    data = get_issue_or_pr_data(event.data['issue'])
+@router.register('pull_request', action='assigned')
+@router.register('pull_request', action='unassigned')
+async def assignment_event(event, gh, *args, **kwargs):
+    if event.data[0] == 'issues': event = 'issue'
```

Comment:
This code "shadows" the event variable, which is passed into the `assignment_event` function, which is not good practice.  You should probably rename the `event` variable something like `event_type`.

Also, these `if` statements can be turned into a one-liner that is easier to read:

```python
event_type = 'issue' if event.data[0] == 'issues' else 'pull_request'
```